pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
+	switch strategy.Type {
+	case deployapi.DeploymentStrategyTypeCustom:
+		util.MergeInto(pod.Labels, strategy.Labels, 0)
```

Comment:
@pedro-r-marques 
>The custom strategy is where the user is supplying the deployer; openshift supplies the deployer in the other strategies and thus (to my knowledge) they do not need custom attributes. I'm happy to change it but i do think it does make sense that as you put it "custom labels/annotations apply to the custom deployer pod".

In all cases, openshift makes the pod and provides some manner of customization over the pod hosting the deployer logic (whether it's from our prefab image or yours). We already support a bit of (indirect) control over the deployer pod generally by allowing the node selector of the pod template to be inherited by the deployer pod... If user added labels/annotations can't break the deployer itself (e.g. by overwriting system keys) then I guess I'm not sure why not just make the new configurations apply to deployers generally. It would be easier to explain and implement.